controller/src/test/java/org/jboss/as/controller/persistence/PersistentResourceXMLParserTestCase.java

```diff
@@ -0,0 +1,209 @@
+                    .setXmlName("attr1")
+                    .build();
+            final AttributeDefinition clusterAttr2 = create("cluster-attr2", ModelType.STRING)
+                    .setAttributeGroup("cluster")
+                    .setXmlName("attr2")
+                    .build();
+
+            final AttributeDefinition securityAttr1 = create("security-my-attr1", ModelType.STRING)
+                    .setAttributeGroup("security")
+                    .setXmlName("my-attr1")
+                    .build();
+            final AttributeDefinition securityAttr2 = create("security-my-attr2", ModelType.STRING)
+                    .setAttributeGroup("security")
+                    .setXmlName("my-attr2")
+                    .build();
+
+
+            RESOURCE_INSTANCE = new PersistentResourceDefinition(PathElement.pathElement("resource"), new NonResolvingResourceDescriptionResolver()) {
+                @Override
+                public Collection<AttributeDefinition> getAttributes() {
+                    Collection<AttributeDefinition> attributes = new ArrayList<>();
+                    attributes.add(clusterAttr1);
+                    attributes.add(clusterAttr1);
+                    attributes.add(securityAttr1);
+                    attributes.add(securityAttr2);
+                    return attributes;
+                }
+            };
+
+
+            PersistentResourceDefinition SUBSYSTEM_ROOT_INSTANCE = new PersistentResourceDefinition(SUBSYSTEM_PATH, new NonResolvingResourceDescriptionResolver()) {
+
+                @Override
+                public Collection<AttributeDefinition> getAttributes() {
+                    return Collections.emptyList();
+                }
+
+                @Override
+                protected List<? extends PersistentResourceDefinition> getChildren() {
+                    List<PersistentResourceDefinition> children = new ArrayList<>();
+                    children.add(RESOURCE_INSTANCE);
+                    return children;
+                }
+            };
+
+            xmlDescription = builder(SUBSYSTEM_ROOT_INSTANCE)
+                    .addChild(
+                            builder(RESOURCE_INSTANCE)
+                                    .addAttributes(
+                                            // cluster group
+                                            clusterAttr1,
+                                            clusterAttr2,
+                                            // security group
+                                            securityAttr1,
+                                            securityAttr2
+                                            )
+                    )
+                    .build();
+        }
+
+        @Override
+        public void readElement(XMLExtendedStreamReader reader, List<ModelNode> list) throws XMLStreamException {
+            xmlDescription.parse(reader, PathAddress.EMPTY_ADDRESS, list);
+        }
+
+        @Override
```

Comment:
no need for this anymore, you should rather extend PersistentResourceXMLParser, where read/write Element methods are hidden.